lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -274,20 +276,22 @@ newStakePoolLayer db@DBLayer{..} nl tr = StakePoolLayer
 
         (distr, prod) <- liftIO . atomically $ (,)
             <$> (Map.fromList <$> readStakeDistribution nodeEpoch)
-            <*> (count <$> readPoolProduction nodeEpoch)
+            <*> readPoolProduction nodeEpoch
+
+        let tip = maximumBy (compare `on` view #slotId)
```

Comment:
Because side-effects are bad :grimacing: ? 
We already have that information available, so there's no need to use another call for this. 

cf my last commit.